st2common/st2common/persistence/base.py

```diff
@@ -112,12 +112,52 @@ def aggregate(cls, *args, **kwargs):
         return cls._get_impl().aggregate(*args, **kwargs)
 
     @classmethod
-    def add_or_update(cls, model_object, publish=True, dispatch_trigger=True):
+    def insert(cls, model_object, publish=True, dispatch_trigger=True,
+               log_not_unique_error_as_debug=False):
+        if model_object.id:
+            raise ValueError('id for object %s was unexpected.' % model_object)
+        try:
+            model_object = cls._get_impl().insert(model_object)
+        except NotUniqueError as e:
+            if log_not_unique_error_as_debug:
+                LOG.debug('Conflict while trying to save in DB.', exc_info=True)
```

Comment:
Yeah, I agree about lack of full traceback being an issue. Having said that, IIRC, if you use traceback module to retrieve the traceback it should also include a stack trace from the previous / underlying exception (unless you explicitly limit it to the last stack frame using limit=1).